### DISCUSSION TITLE
How to test that telemetries are disabled?

### DISCUSSION THREAD

**Discussion by michalperry** [microsoft/onnxruntime#16879] [Other Q&A] [answered]
Hi community,
We are planning to integrate onnxruntime (v1.12.0) in our product (cpp based) for xgboost model inference.
From a security point we are concerned about the telemetries sent outside. 
In the docs it is stated that telemetries can be disabled using the method - 'DisableTelemetryEvents()'

Still, how can we make sure that disabling the telemetries sending **indeed disable any outgoing packets?**

I tried 'wireshark' software to identify packets going out during inference but couldn't identify any packets either when disable or enable the telemetries.

Identifying the place in code that disables the telemetry can be a last resort. Need help to point out the place in the source code where telemetries are disabled.

Thanks in advance

**Comment by xadupre** [comment]
Which OS are you using?

**Reply by michalperry** [reply]
product is targeted for both Windows and linux

**Reply by snnn** [reply]
We do not have telemetry on Linux. 
On Windows it is part of Windows telemetry. You need to refer Windows doc to know how to disable it(it is not always possible). 
Or you can build ONNX Runtime from source. By default the code doesn't use a meaningful telemetry option, you can see it is all zero:
https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/platform/windows/TraceLoggingConfig.h#L26
Our prebuilt binaries were built with a different telemetry option.  And surely, if you don't like it you can totally delete all the telemetry related code, it won't affect any usability.

**Reply by michalperry** [reply]
thanks, this completes the picture of how it works

**Comment by xadupre** [comment, marked as answer]
Telemetry is implemented here: https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/platform/windows/telemetry.cc.

**Reply by michalperry** [reply]
@xadupre Thanks a lot, this is what I was looking for.

**Comment by xadupre** [comment]
Telemetry is not implemented on Linux. On Windows, this page https://github.com/microsoft/onnxruntime/blob/main/docs/Privacy.md describes how the telemetry works when it is enabled.
